packages/web/modals/token-select-modal-limit.tsx

```diff
@@ -0,0 +1,535 @@
+import { PricePretty } from "@keplr-wallet/unit";
+import { DEFAULT_VS_CURRENCY } from "@osmosis-labs/server";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import Image from "next/image";
+import {
+  FunctionComponent,
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { useLatest } from "react-use";
+
+import { Icon } from "~/components/assets";
+import { Intersection } from "~/components/intersection";
+import { Spinner } from "~/components/loaders";
+import {
+  useFilteredData,
+  useTranslation,
+  useWalletSelect,
+  useWindowKeyActions,
+} from "~/hooks";
+import { useConst } from "~/hooks/use-const";
+import { useDraggableScroll } from "~/hooks/use-draggable-scroll";
+import { useKeyActions } from "~/hooks/use-key-actions";
+import { useStateRef } from "~/hooks/use-state-ref";
+import { SwapAsset, useRecommendedAssets } from "~/hooks/use-swap";
+import { ActivateUnverifiedTokenConfirmation, ModalBase } from "~/modals";
+import { useStore } from "~/stores";
+import { UnverifiedAssetsState } from "~/stores/user-settings";
+import { formatPretty } from "~/utils/formatter";
+
+const dataAttributeName = "data-token-id";
+
+function getTokenItemId(uniqueId: string, index: number) {
+  return `token-selector-item-${uniqueId}-${index}`;
+}
+
+function getTokenElement(uniqueId: string, index: number) {
+  return document.querySelector(
+    `[${dataAttributeName}=${getTokenItemId(uniqueId, index)}]`
+  );
+}
+
+function getAllTokenElements() {
+  return document.querySelectorAll(`[${dataAttributeName}]`);
+}
```

Comment:
BTW we have a useKeyboardNavigation hook that encapsulates this functionality. It may clean things up a bit.

Reply:
Done here:
https://github.com/osmosis-labs/osmosis-frontend/pull/3683/commits/8dae53ae7981c093de4783ca6a62c52262be68e8